core/graphelier-service/api/hndlrs/msghandler.go

```diff
@@ -0,0 +1,52 @@
+package hndlrs
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"graphelier/core/graphelier-service/models"
+
+	"github.com/gorilla/mux"
+)
+
+// FetchMessages returns messages given an instrument and sod_offset with support for
+// pagination
+func FetchMessages(env *Env, w http.ResponseWriter, r *http.Request) error {
+	nMessages := r.URL.Query().Get("nMessages")
+
+	params := mux.Vars(r)
+	instrument := params["instrument"]
+	sodOffset := params["sodOffset"]
+
+	intNMessages, err := strconv.ParseInt(nMessages, 10, 8)
+	if err != nil {
+		intNMessages = 20
+	}
+
+	intSodOffset, err := strconv.ParseInt(sodOffset, 10, 64)
+	if err != nil {
+		return StatusError{400, err}
+	}
+	paginator := models.Paginator{NMessages: intNMessages, SodOffset: intSodOffset}
+	messages, err := env.Connector.GetMessagesWithPagination(instrument, intSodOffset, &paginator)
+	if err != nil {
+		return StatusError{500, err}
+	}
+	messagePage := models.MessagePage{PageInfo: paginator, Messages: messages}
+	if intNMessages < 0 {
+		reverseMessageOrdering(&messagePage)
+	}
+	w.WriteHeader(http.StatusOK)
+	err = json.NewEncoder(w).Encode(messagePage)
+	if err != nil {
+		return StatusError{500, err}
+	}
```

Comment:
is it good practice to constantly re-affect by overriding the `err` var in this block? or creating multiple `err` variables with the same name, since the scope isn't quite defined

Reply:
I believe this is proper Go style for error handling